### ISSUE TITLE
Annotations Graphics Export doesn't export for all features

### ISSUE THREAD

**Issue by AndrewDGillen** [mzmine/mzmine#2301]
### Basic information

* _My operating system and version:_ Ubuntu 24.04.1
* _My mzmine version:_ 4.4.3

### What happened

 _Please describe what happened and how it differs from what you expected to happen._

I've been setting up a pipeline for running batch analyses, and as part of that I was hoping to export all chromatograms from identified features. For this, I've been using the Annotations Graphics Export module, but unfortunately it seems to only export chromatograms from a subset of identified features (in our case, for 28 of 5822 features), seemingly only those with a spectral match. 

From looking at the code for the Annotations Graphics Export module, it's possible this is the expected behavior, but I can't find a great work around for my use case. Is there any way to export chromatograms for all features from a feature list when running in batch mode?

A full project file, along with a batch file used for running and the associated log, can be found at:
https://drive.google.com/drive/folders/1QdfkjhovwUPCJL98r9BKqxFfaFcRJk4L?usp=sharing

**Comment by SteffenHeu** [comment]
Hi Andrew, 

yes you're right, it's the expected behaviour to only export annotated features.

What's the reason you want to export the chromatograms for all features? E.g. Downstream applications? Perhaps there is a better solution than exporting graphics?

**Comment by AndrewDGillen** [comment]
Hi Steffen,

Thanks for getting back to me so quickly - we'd like to be able to see the actual peaks associated with each feature for QC and potential further manual annotation.  

Even aside from exporting non-annotated peaks, we're not able to export chromatograms from features which have been annotated by other means aside from spectral matching (eg, local database matches). If we're reporting that we've found something in the mass spec data, it'd be really nice to be able to show the peak alongside that.

**Comment by SteffenHeu** [comment]
You are right, the compound matches are not included at the moment, we will add those.

As for exporting all features - i think the overhead for that would be too big to actually be useful. Are you processing on a server and have no access to the gui?

**Comment by AndrewDGillen** [comment]
A server with no GUI access would be the plan, yeah.

Since I posted the issue I've tried manually removing the code limiting the export to only feature rows with lipid/spectral match annotations and that seemed to run quite happily locally, producing chromatograms for all features on a machine with 32 Gb RAM.

I can certainly understand that for most applications it's kind of overkill to export everything, but might it be an option to let users pick and choose which subset of features to export?

**Comment by AndrewDGillen** [comment]
That'd be perfect, thanks for your help!

**Comment by AndrewDGillen** [comment]
I have a new, somewhat related issue - Graphical export fails to complete in headless mode. I hadn't realized this would be an issue since I was setting up the pipeline with the GUI.

I've updated my google drive (https://drive.google.com/drive/folders/1QdfkjhovwUPCJL98r9BKqxFfaFcRJk4L?usp=sharing) with relevant project files/logs, but it seems like there's an issue in starting up JavaFX threads - see screenshot below taken from the log.

![image](https://github.com/user-attachments/assets/f1d9418e-1c37-473b-87e1-98731d0a011f)

Thanks again for all your help on this!
